### PR TITLE
Make the JXL Makefile work with both tags and SHAs.

### PR DESCRIPTION
**Pull request by veluca93**
This allows to update the codec without relying on upstream tags.

I didn't make the checkout rule depend on the Makefile, which means that version updates will only come into effect after removing `node_modules`, but it should be possible in principle.

### CODE REVIEW

File: codecs/jxl/Makefile
Comment (deymo) [inline codecs/jxl/Makefile]: `git init $(@D)`

and then you can do `git -C $(@D) remote add ......` so you don't need to do `cd`

File: codecs/jxl/Makefile
Comment (deymo) [inline codecs/jxl/Makefile]: I'd checkout `FETCH_HEAD` here since the "$(CODEC_VERSION)" may resolve to something different here. For example, if CODEC_VERSION is "master" this will resolve first to the local "master" branch if one exists rather than origin/master.

**Comment by surma** [comment]
I actually think we used to have this and we changed it for some reason. @RReverser do you remember?

Either way, this LGTM. Gonna wait for Ingvar to make the final call, tho.

**Comment by veluca93** [comment]
> I actually think we used to have this and we changed it for some reason. @RReverser do you remember?
> 
> Either way, this LGTM. Gonna wait for Ingvar to make the final call, tho.

I think the reason was that the old way didn't do shallow checkouts and/or parallel cloning of submodules (don't remember which, possibly both), but this implementation ought to do both.

**Comment by RReverser** [comment]
> I actually think we used to have this and we changed it for some reason. @RReverser do you remember?

Yeah, I added it on my JPEG-XL SIMD branch but then reverted before merging the PR because a new version was tagged, and it didn't seem worth the Makefile complication anymore.

Do we think it's still / again worth it? If there's a usecase (e.g. tags are not catching up fast enough), I don't mind.

> I think the reason was that the old way didn't do shallow checkouts and/or parallel cloning of submodules (don't remember which, possibly both), but this implementation ought to do both.

The one we're talking about, did - it was pretty similar to your PR (I think the only difference is `-C $(@D)` vs `cd ... && git ... && git ...`).

File: codecs/jxl/Makefile
Comment (RReverser) [inline codecs/jxl/Makefile]: This line is unnecessary, you can just `fetch $(CODEC_URL) $(CODEC_VERSION)` below.

Comment (veluca93) [reply codecs/jxl/Makefile]: Nice!

**Comment by veluca93** [comment]
> > I actually think we used to have this and we changed it for some reason. @RReverser do you remember?
> 
> Yeah, I added it on my JPEG-XL SIMD branch but then reverted before merging the PR because a new version was tagged, and it didn't seem worth the Makefile complication anymore.
> 
> Do we think it's still / again worth it? If there's a usecase (e.g. tags are not catching up fast enough), I don't mind.
> 
> > I think the reason was that the old way didn't do shallow checkouts and/or parallel cloning of submodules (don't remember which, possibly both), but this implementation ought to do both.
> 
> The one we're talking about, did - it was pretty similar to your PR (I think the only difference is `-C $(@D)` vs `cd ... && git ... && git ...`).

Yup, it's easier to get bugfixes by checking out a sha instead of waiting for a release on our side (which is actually slightly more involved than just tagging a commit)

File: codecs/jxl/Makefile
Comment (RReverser) [inline codecs/jxl/Makefile]:
```suggestion
	$(RM) $(@D)
```

File: codecs/jxl/Makefile
Comment (RReverser) [inline codecs/jxl/Makefile]: Found one more optimisation that was in my fork but not here:

```suggestion
	git -C $(@D) submodule update --init --depth 1 --recursive --jobs `nproc`
```

File: codecs/jxl/Makefile
Comment (veluca93) [reply codecs/jxl/Makefile]: This still needs a -r, right? Done.

**Review by RReverser — APPROVED** [review]

**Pull request merged**
